src/ui/repo-tree-model.cpp

```diff
@@ -61,8 +61,11 @@ RepoTreeModel::RepoTreeModel(QObject *parent)
 }
 RepoTreeModel::~RepoTreeModel()
 {
-    if (item(kIndexOfVirtualReposCategory) != virtual_repos_category_)
-        delete virtual_repos_category_;
+    // if we miss it, just add it and
+    // let dtor of repotreemodel to delete it automatically
+    if (item(kIndexOfVirtualReposCategory) != virtual_repos_category_) {
+        insertRow(kIndexOfVirtualReposCategory, virtual_repos_category_);
+    }
```

Comment:
Doing this(inserting a row) in the destructor really feels counter-intuitive to me.